src/ott/neural/models/base_solver.py

```diff
@@ -0,0 +1,485 @@
+      )
+      return ot_solver(
+          linear_problem.LinearProblem(geom, tau_a=tau_a, tau_b=tau_b)
+      ).matrix
+
+    return match_pairs_filtered if filter_input else match_pairs
+
+  def _get_gromov_match_fn(
+      self,
+      ot_solver: Any,
+      cost_fn: Union[Any, Mapping[str, Any]],
+      scale_cost: Union[Union[bool, int, float,
+                              Literal["mean", "max_norm", "max_bound",
+                                      "max_cost", "median"]],
+                        Dict[str, Union[bool, int, float,
+                                        Literal["mean", "max_norm", "max_bound",
+                                                "max_cost", "median"]]]],
+      tau_a: float,
+      tau_b: float,
+      fused_penalty: float,
+  ) -> Callable:
+    if isinstance(cost_fn, Mapping):
+      assert "cost_fn_xx" in cost_fn
+      assert "cost_fn_yy" in cost_fn
+      cost_fn_xx = cost_fn["cost_fn_xx"]
+      cost_fn_yy = cost_fn["cost_fn_yy"]
+      if fused_penalty > 0:
+        assert "cost_fn_xy" in cost_fn_xx
+        cost_fn_xy = cost_fn["cost_fn_xy"]
+    else:
+      cost_fn_xx = cost_fn_yy = cost_fn_xy = cost_fn
+
+    if isinstance(scale_cost, Mapping):
+      assert "scale_cost_xx" in scale_cost
+      assert "scale_cost_yy" in scale_cost
+      scale_cost_xx = scale_cost["scale_cost_xx"]
+      scale_cost_yy = scale_cost["scale_cost_yy"]
+      if fused_penalty > 0:
+        assert "scale_cost_xy" in scale_cost
+        scale_cost_xy = cost_fn["scale_cost_xy"]
+    else:
+      scale_cost_xx = scale_cost_yy = scale_cost_xy = scale_cost
+
+    @jax.jit
+    def match_pairs(
+        x_lin: Optional[jnp.ndarray],
+        x_quad: Tuple[jnp.ndarray, jnp.ndarray],
+        y_lin: Optional[jnp.ndarray],
+        y_quad: Tuple[jnp.ndarray, jnp.ndarray],
+    ) -> Tuple[jnp.array, jnp.array]:
+      geom_xx = pointcloud.PointCloud(
+          x=x_quad, y=x_quad, cost_fn=cost_fn_xx, scale_cost=scale_cost_xx
+      )
+      geom_yy = pointcloud.PointCloud(
+          x=y_quad, y=y_quad, cost_fn=cost_fn_yy, scale_cost=scale_cost_yy
+      )
+      if fused_penalty > 0:
+        geom_xy = pointcloud.PointCloud(
+            x=x_lin, y=y_lin, cost_fn=cost_fn_xy, scale_cost=scale_cost_xy
+        )
+      else:
+        geom_xy = None
+      prob = quadratic_problem.QuadraticProblem(
+          geom_xx,
+          geom_yy,
+          geom_xy,
+          fused_penalty=fused_penalty,
+          tau_a=tau_a,
+          tau_b=tau_b
+      )
+      out = ot_solver(prob)
+      return out.matrix
+
+    return match_pairs
+
+
+class UnbalancednessMixin:
+  """Mixin class to incorporate unbalancedness into neural OT models."""
+
+  def __init__(
+      self,
+      rng: jax.Array,
+      source_dim: int,
+      target_dim: int,
+      cond_dim: Optional[int],
+      tau_a: float = 1.0,
+      tau_b: float = 1.0,
+      rescaling_a: Optional[Callable[[jnp.ndarray, Optional[jnp.ndarray]],
```

Comment:
Is there any scenario where the user doesn't want to learn the `rescaling_a` or `rescaling_b`?

Reply:
The other question I have: would it make sense to have only 1 class (that learns either to rescale a or b) and later instantiate the class twice if the user wants to learn how to rescale both a and b?

Reply:
Yes, I think one does not necessarily want to learn the rescaling factors, as it requires memory and time. For most of the applications we don't learn it in https://openreview.net/forum?id=2UnCj3jeao